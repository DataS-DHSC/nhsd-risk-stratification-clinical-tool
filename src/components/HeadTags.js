import Head from 'next/head';
import PropTypes from 'prop-types';
import { SITE_BRANDING_TITLE } from '../constants/site-branding';
import AdobeAnalytics from './analytics/AdobeAnalytics';

const HeadTags = ({ title, description, featureFlags }) => (
  <>
    <Head>
      <title>
        {SITE_BRANDING_TITLE} {title ? ` - ${title}` : ''}
      </title>
      <meta
        name="description"
        content={`${SITE_BRANDING_TITLE} ${description}`}
      />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>

    <AdobeAnalytics featureFlags={featureFlags} />
  </>
);

HeadTags.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  featureFlags: PropTypes.shape(),
};

HeadTags.defaultProps = {
  title: undefined,
  description: undefined,
  featureFlags: {},
};

export default HeadTags;
