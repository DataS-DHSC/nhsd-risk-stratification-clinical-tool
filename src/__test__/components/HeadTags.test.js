import React from 'react';
import { render } from '@testing-library/react';
import HeadTags from '../../components/HeadTags';

const setup = () => {
  const props = {
    title: 'Start page',
    description: 'description',
  };

  return render(<HeadTags {...props} />);
};

describe('HeadTags', () => {
  it('should match snapshot ', () => {
    const container = setup();
    expect(container.asFragment()).toMatchSnapshot();
  });
});
