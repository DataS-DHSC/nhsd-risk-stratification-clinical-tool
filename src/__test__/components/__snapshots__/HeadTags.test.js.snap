// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadTags should match snapshot  1`] = `<DocumentFragment />`;
